File: pset6/caesar.py
import sys

# make sure argument count is 2 and second argument (key) is int
if len(sys.argv) == 2 and sys.argv[1].isdecimal():
    key = int(sys.argv[1]) % 26
else:
    print ("Usage: python caesar.py [int]")
    sys.exit(1)

# input must not be empty
while True:
    plain = input("plaintext: ")
    if plain != "":
        break

print ("ciphertext: ", end="")    
    
# list letters in order
uppers = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z']
lowers = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
    
# iterate over characters;
for c in plain:
    if c.isalpha():
        if c.isupper():
            # lower case character - 65 is index on list; add key; mod 26 to wrap around
            letter = (uppers[(ord(c) - 65 + key) % 26])
            print (letter, end="")
        elif c.islower():
            # upper case character - 97 is index on list; add key; mod 26 to wrap around
            letter = (lowers[(ord(c) - 97 + key) % 26])
            print (letter, end="")
    else:
        print (c, end="")
        
print("")

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    // check if argc = 2, else return 1
    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    
    printf("plaintext:  ");
    string p = get_string();
    // convert string argv[1] to int; if over 26, mod 26
    int k = atoi(argv[1]);
    if (k > 26)
    {
        k %= 26;
    }
    if (p != NULL)
    printf("ciphertext: ");
    {
        // iterate over characters in string
        for (int i = 0, n = strlen(p); i < n; i++)
            {
                // check if p[i] is a letter 
                if (isalpha(p[i]))
                {
                    // make letter wrap around if necessary
                    if ((p[i] <= 90 && p[i] + k > 90) || (p[i] <= 122 && p[i] + k > 122))
                    {
                        p[i] -= 26;
                    }
                printf("%c", p[i] + k);
                }
                // if not a letter, don't use key
                else
                printf("%c", p[i]);
            }
        printf("\n");
    }
    return 0;
}